### DISCUSSION TITLE
Consulta auto

### DISCUSSION THREAD

**Discussion by andromeda2075** [cpp-review-dune/introductory-review#16] [Q&A]
https://asciinema.org/a/470253

**Comment by carlosal1015** [comment]
Supongamos que tenemos estos dos archivos
![CMakeLists](https://user-images.githubusercontent.com/21283014/154875845-71409d89-be67-4200-8057-5978b08d504a.png)
![main](https://user-images.githubusercontent.com/21283014/154875903-b96b2956-ea8b-4152-993a-0e417d4cc8a4.png)
![image](https://user-images.githubusercontent.com/21283014/154876418-4a2f7676-391d-42a5-9ff5-1d18f9bb27ad.png)

 y lo ejecutamos con `cmake -S . -B build`

En verdad, en las líneas 5 y 11 la conversión es para un entero, en la linea 8 la conversión es para un entero largo. Sin embargo, en la línea 15 la conversión no es para un entero, sino una lista (un tipo especial de [contenedor](https://en.cppreference.com/w/cpp/container)), por lo cual debemos recorrerlo.

- [New Rules for auto deduction from braced-init-list](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n3922.html)
